.github/workflows/ci-tests.yml

```diff
@@ -54,17 +51,18 @@ jobs:
         uses: KyleMayes/install-llvm-action@8852e4d5c58653ed05135c0a5d949d9c2febcb00 # v1.6.1
         with:
           version: "15.0"
-        if: runner.os == 'macOS'
+        if: contains(fromJSON('["macOS", "macos-12", "Windows", "windows-latest", "windows-2022"]'), runner.os)
```

Comment:
I hope we can make our runners to have the same value of `runner.os` as GitHub's. What you have here looks like a wild guess, makes no sense to have such a list of things, just makes no sense. For instance `windows-latest` will never ever match this.

As comment says the only reason we have thing step in the first place is because on macOS runners Clang is Apple's version and we need upstream with wasm32 support. On all other platforms GitHub's runners already have upstream version installed and don't need this step (will just waste CI time).

Also you're likely confusing `runner.os` with `matrix.os`.